rust/ares/src/hamt.rs

```diff
@@ -437,17 +437,26 @@ impl<T: Copy + Preserve> Preserve for Hamt<T> {
             let dest_buffer = stack.struct_alloc_in_previous_frame(self.0.size());
             copy_nonoverlapping(self.0.buffer, dest_buffer, self.0.size());
             self.0.buffer = dest_buffer;
-            *(stack.push::<(Stem<T>, u32)>()) = (self.0, 0);
+            // Here we're using the Rust stack since the array is a fixed
+            // size. Thus it will be cleaned up if the Rust thread running
+            // this is killed, and is therefore not an issue vs. if it were allocated
+            // on the heap.
+            //
+            // In the past, this traversal stack was allocated in NockStack, but
+            // exactly the right way to do this is less clear with the split stack.
+            let mut traversal_stack: [Option<(Stem<T>, u32)>; 6] = [None; 6];
+            traversal_stack[0] = Some((self.0, 0));
             let mut traversal_depth = 1;
             'preserve: loop {
                 if traversal_depth == 0 {
                     break;
                 }
-                let (stem, mut position) = *(stack.top::<(Stem<T>, u32)>());
+                let Some((stem, mut position)) = traversal_stack[traversal_depth - 1] else {
+                    panic!("Attempted to access uninitialized array element");
+                };
```

Comment:
```suggestion
                let (stem, mut position) = traversal_stack[traversal_depth - 1].expect("Attempted to access uninitialized array element");
```

`.expect()` is the canonical way to do this for both `Result` and `Option`